docs/quaternions.rst

```diff
@@ -0,0 +1,264 @@
+
+Quaternion Quick Facts
```

Comment:
I have not re-verified the math here.  It is a great idea to have it but not sure if it will get peer review.